cli/flox/src/commands/services/mod.rs

```diff
@@ -101,6 +102,22 @@ pub fn supported_environment(
     Ok(dyn_environment)
 }
 
+/// Warn about manifest changes that may require services to be restarted, if
+/// the Environment has a service manager running. It doesn't guarentee that the
+/// service manager is working (e.g. hasn't crashed). It is the caller's
+/// responsibility to determine what manifest changes would affect services.
+pub fn warn_manifest_changes_for_services(flox: &Flox, env: &dyn Environment) {
```

Comment:
> The same changes for `flox pull` will come in a separate PR.

I haven't quite figured out how this is going to work yet because `pull_existing_environment` has a `ManagedEnvironment` rather than an `Environment`. We'd have to move some of the work to get the socket path up to the callers.

Reply:
Ahh, I had misunderstood and not actually tried it on Friday.

I'll still follow-up with `pull` in a separate PR though because the testing is slightly more complicated and I'd like to get these changes into the release.